e2e_test/batch/functions/array_concat.slt.part

```diff
@@ -124,12 +29,12 @@ select array[1,2] || array[array[3,4]];
 {{1,2},{3,4}}
 
 query T
-select '123'::varchar || array['abc'];
+select ('123' || '') || array['abc'];
```

Comment:
`select '123' || array['abc'];` fails in PostgreSQL but `select '123'::varchar || array['abc'];` is okay.
We treat `'123'` and `'123'::varchar` the same and rejects both right now, due to the fix of `'{123}' || array['abc']`.
The `|| ''` is a workaround (similar to how PostgreSQL requires extra `::varchar`) and we would fix to PostgreSQL behavior later.